cmd/otk/osbuild-resolve-ostree-commit/main.go

```diff
@@ -27,6 +27,16 @@ type Input struct {
 
 	// Whether to use RHSM secrets when resolving and fetching the commit.
 	RHSM bool `json:"rhsm,omitempty"`
+
+	// MTLS information. Will be ignored if RHSM is set.
```

Comment:
There is a `main_test.go:TestMockResolve()` test, would it make sense to add the new inputs there (or create a new test analog to the existing one) to ensure we do not regress here?

Reply:
Sure added but I do not understand why tests are failing now.

Reply:
Was looking at the wrong branch, rebased :-D

Reply:
Well, tests are failing (locally too) :/
```console
$ go test -run TestResolver
--- FAIL: TestResolver (0.00s)
    main_test.go:90: 
        	Error Trace:	/home/mvogt/devel/osbuild/images/cmd/otk/osbuild-resolve-ostree-commit/main_test.go:90
        	Error:      	Not equal: 
        	            	expected: map[string]map[string]map[string]string{"tree":map[string]map[string]string{"const":map[string]string{"checksum":"9b1ea9a8e10dc27d4ea40545bec028ad8e360dd26d18de64b0f6217833a8443d", "ref":"test/ref/alpha", "url":"http://127.0.0.1:45097"}}}
        	            	actual  : map[string]map[string]map[string]string{"tree":map[string]map[string]string{"const":map[string]string{"checksum":"9b1ea9a8e10dc27d4ea40545bec028ad8e360dd26d18de64b0f6217833a8443d", "ref":"test/ref/alpha", "secrets":"org.osbuild.mtls", "url":"http://127.0.0.1:45097"}}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,5 +2,6 @@
        	            	  (string) (len=4) "tree": (map[string]map[string]string) (len=1) {
        	            	-  (string) (len=5) "const": (map[string]string) (len=3) {
        	            	+  (string) (len=5) "const": (map[string]string) (len=4) {
        	            	    (string) (len=8) "checksum": (string) (len=64) "9b1ea9a8e10dc27d4ea40545bec028ad8e360dd26d18de64b0f6217833a8443d",
        	            	    (string) (len=3) "ref": (string) (len=14) "test/ref/alpha",
        	            	+   (string) (len=7) "secrets": (string) (len=16) "org.osbuild.mtls",
        	            	    (string) (len=3) "url": (string) (len=22) "http://127.0.0.1:45097"
        	Test:       	TestResolver
    main_test.go:90: 
        	Error Trace:	/home/mvogt/devel/osbuild/images/cmd/otk/osbuild-resolve-ostree-commit/main_test.go:90
        	Error:      	Not equal: 
        	            	expected: map[string]map[string]map[string]string{"tree":map[string]map[string]string{"const":map[string]string{"checksum":"7433e1b49fb136d61dcca49ebe34e713fdbb8e29bf328fe90819628f71b86105", "ref":"test/ref/one", "url":"http://127.0.0.1:45097"}}}
        	            	actual  : map[string]map[string]map[string]string{"tree":map[string]map[string]string{"const":map[string]string{"checksum":"7433e1b49fb136d61dcca49ebe34e713fdbb8e29bf328fe90819628f71b86105", "ref":"test/ref/one", "secrets":"org.osbuild.mtls", "url":"http://127.0.0.1:45097"}}}
...
```